pkg/workspace/inference/preset-inferences.go

```diff
@@ -193,16 +209,67 @@ func CreatePresetInference(ctx context.Context, workspaceObj *v1beta1.Workspace,
 	image, imagePullSecrets := GetInferenceImageInfo(ctx, workspaceObj, inferenceParam)
 
 	var depObj client.Object
-	if model.SupportDistributedInference() {
-		depObj = manifests.GenerateStatefulSetManifest(workspaceObj, image, imagePullSecrets, *workspaceObj.Resource.Count, commands,
+	if model.SupportDistributedInference() && runtimeName == pkgmodel.RuntimeNameVLLM {
+		// 60 seconds initial delay for liveness probe to allow workers to join the cluster
+		livenessProbe := getDistributedInferenceProbe(probeTypeLiveness, workspaceObj, 60, 10, 5)
+		readinessProbe := getDistributedInferenceProbe(probeTypeReadiness, workspaceObj, 0, 10, 1)
+		depObj = manifests.GenerateStatefulSetManifest(workspaceObj, image, imagePullSecrets, numNodes, commands,
```

Comment:
I can see if numNodes == 1, you will still use statefulset to place this workload. It's good from the sight of implementation and vllm will still use ray for single node inference.

I have a consideration here. If we always use deployment for single-node inference and lws for multi-node inference, it would be easier for autoscaler to scale and manage workload. WDYT?

Reply:
SGTM